scio-smb/src/main/scala/com/spotify/scio/smb/util/SMBMultiJoin.scala

```diff
@@ -23,11 +23,14 @@ import com.spotify.scio.ScioContext
 import com.spotify.scio.coders.Coder
 import com.spotify.scio.values._
 import org.apache.beam.sdk.extensions.smb.{SortedBucketIO, TargetParallelism}
+import com.spotify.scio.smb.syntax.SortedBucketScioContext
 
 import scala.jdk.CollectionConverters._
 
 final class SMBMultiJoin(private val self: ScioContext) {
 
+  val sortedBucketScioContext = new SortedBucketScioContext(self)
```

Comment:
this should be `private`. Though, I think `SMBMultiJoin` should have dependency only on `SortedBucketScioContext` avoiding this. It's possible to change it and still keep backwards compat If I'm looking at this correctly.

Reply:
Thanks for the feedback @regadas!
Since the SortedBucketScioContext class is final, I couldn't extend it so had to initialize it here. Am I missing something here? Can you please explain what changes can I make here?

Reply:
I don't think removing the `final` keyword from `SortedBucketScioContext` would affect binary compatibility (at least in Java...@regadas maybe you know differently for Scala).

Reply:
Sorry for the late reply on this! yeah removing `final` wouldn't affect it; here we also don't want to extend the `SortedBucketScioContext`.

After a second look, let's keep like this but making it `private`. Since it's my fault that this is delayed I'll follow up with the changes.